docs/enhancements/ai-on-zero-worker-hypershift.md

```diff
@@ -0,0 +1,121 @@
+
+#### Selected solution
+<b>New CRD HypershiftServiceConfig on a namespace scope<b>
+- Separation of intent
+- Natural namespacing
+
+### Assisted-service controllers working with hostedCluster (L1) control plane
+From the user point of view, they define the cluster installation CRs on one of the namespaces in the hypershift cluster, and the assisted service picks them up and spins the cluster. To achieve that, we need to:
+1. Have all cluster deployment CRDs defined on L1 
+2. Assisted service's controllers should work with L1 api-server (the hosted cluster) exclusively, even though the pods on which the controllers are running are hosted on L0 (the management cluster).
+
+#### CRD deployment
+This task, which is currently done by the OLM on the management cluster, should now be carried out by the operator.
+The HASC controller on L0 is going to list all the assisted-service-related CRDs currently defined on the L0 cluster. Those CRDs are placed there by the OLM itself. So in this way, we are ensuring compatibility with the non-hypershift installation. 
+After the CRD BOM is collected, the HASC operator going to use the L1 client to define the same CRDs on L1's control plane.
+### HypershiftAgentServiceConfig controller (Infrastructure operator)
+#### Client configuration
+KUBECONFIG env variable on the service deployment points to a mounted secret containing L1's kubeconfig. This secret is defined in a well-defined place in the management cluster and is also used by ACM. The mounted secret is specified by the KubeconfigSecretRef field in HypershiftAgentServiceConfig CR.
+
+The controller shall register itself to watch and manage resources with both L0 and L1 api-servers and shall instantiate 2 clients, one for each control plane.
+
+Open Issue: How and where is that secret provisioned and become known to us (AI/ACM integration issue)
+
+#### Separating L0 and L1 resources
+The resources to watch will be created by a factory and divided into L0 and L1 resources. 
+
+The infrastructure controller then uses the appropriate clients and the inventory of resources to provision and update the necessary configuration according to the scheme above.
+
+#### RBAC
+On L1 we need the following permissions:
+- To support admission webhook configuration, add a ClusterRole to access validating/mutating webhook configurations, subjectaccessreviews
+- To support the service's controllers, add a ClusterRole to access secrets, routes, nodes, pods, endpoints, services, etc.
+
+This is a mirroring of the RBAC state on L0 in non-hypershift setup.
+
+On L0 we need a few permission that previously were defined by the OLM to allow the service to operate:
+- To support the service's leader election mechanism a Role to access leases and configmaps
+
+Open Issue: Can the leader-election-related roles be defined only on L1?
```

Comment:
We still need to verify it. In the mean time it is an open issue

Reply:
I'm confident it will work, but testing to verify is of course reasonable. I wrote one of the two common leader election implementations and am very familiar with the other.